compiler/rustc_typeck/src/collect/type_of.rs

```diff
@@ -548,9 +555,12 @@ fn find_opaque_ty_constraints(tcx: TyCtxt<'_>, def_id: LocalDefId) -> Ty<'_> {
             }
             // Use borrowck to get the type with unerased regions.
             let concrete_opaque_types = &self.tcx.mir_borrowck(def_id).concrete_opaque_types;
-            if let Some((opaque_type_key, concrete_type)) =
-                concrete_opaque_types.iter().find(|(key, _)| key.def_id == self.def_id)
-            {
+            for (opaque_type_key, concrete_type) in concrete_opaque_types {
+                if opaque_type_key.def_id != self.def_id {
+                    // Ignore constraints for other opaque types.
+                    continue;
```

Comment:
Can't we do

```rust
for (opaque_type_key, concrete_type) in concrete_opaque_types
    .iter()
    .filter(|(key, _)| key.def_id == self.def_id) {
```

Reply:
Nope, `self.def_id` would borrow `self` immutably, so we can't assign to it in the loop itself. (Also I personally don't like mixing iterator combinators and for loops too much, too much back and forth)

Reply:
Ohh right.